### DISCUSSION TITLE
Self-hosted SwaggerUI URL

### DISCUSSION THREAD

**Discussion by stnokott** [helldivers-2/api#72] [Q&A] [answered]
Hello,

possibly a very simple question since I appear to be a moron:

**What is the URL of the SwaggerUI when self-hosting the Dockerfile?**

`http://localhost:8080/api/docs/openapi/swagger-ui.html` (like the GitHub pages path) yields a 404...

And yes, the base path is correct, since `http://localhost:8080/raw/api/WarSeason/current/WarID` works just fine.

**Comment by dealloc** [comment]
I think one of the downsides to #52 is that the API itself doesn't include it's OpenAPI specification anymore.
While this is done for generating the docs (over on GH pages), this does make it annoying to work with self hosted docker containers.

This is definitely something we need to address

**Comment by stnokott** [comment]
For anyone finding this:

Until the issue is addressed, you can build a _very rough_ local replica of the GitHub pages deployment using a custom Dockerfile:

```Dockerfile
FROM mcr.microsoft.com/dotnet/sdk:8.0-alpine AS build

# Install DocFX
RUN dotnet tool install --tool-path=/tools docfx

RUN git clone https://github.com/helldivers-2/api

WORKDIR /api

# Initial static JSON schema submodule
RUN git submodule update --init ./src/Helldivers-2-Models/json

# Generate OpenAPI JSON files
RUN dotnet build -c Debug

# Generate DocFX metadata
RUN /tools/docfx metadata

# Generate DocFX site
RUN /tools/docfx build docfx.json

FROM nginx:stable-alpine AS final
COPY --from=build ["/api/_site", "/usr/share/nginx/html"]
EXPOSE 80
```

1. Build with `docker build -t helldivers2-docs .`
2. Deploy with `docker run --rm -p 80:80 helldivers2-docs`
3. Access the SwaggerUI at `http://localhost:80/docs/openapi/swagger-ui.html`

**Reply by dealloc** [reply, marked as answer]
good news @stnokott I've added OpenAPI support in #73
could you check out that feature branch locally and see if running the containers as documented works?

to save you some clicks, once you're on the feature branch (pending merge) you should be able to build a container with OpenAPI support like this:
```shell
docker build --build-arg="OPENAPI=true" -f ./src/Helldivers-2-API/Dockerfile -t helldivers2-api .
```

**Reply by dealloc** [reply]
I did just notice that the `OPENAPI` flag does not enable SwaggerUI, just the OpenAPI specs, if that's a blocker for you feel free to indicate so in #73!

That being said, you should definitely be able to load the OpenAPI files in any SwaggerUI/editor

**Reply by stnokott** [reply]
Wow, swift reply!

Great to have the OpenAPI spec directly within the same container!
Added a comment https://github.com/helldivers-2/api/pull/73#issuecomment-2051737347 to the MR though as it would be even greater to have the SwaggerUI as well.

Let's continue the discussion there.
